Proofs/SHA512/Sha512_block_armv8.lean

```diff
@@ -80,6 +81,25 @@ theorem sha512_program_test_2_sym (s0 s_final : ArmState)
 
 -- Test 3:
 
+theorem Aligned_BitVecAdd_64_4 {x : BitVec 64} {y : BitVec 64}
+  (x_aligned : Aligned x 4)
+  (y_aligned : Aligned y 4)
+  :
+  Aligned (x + y) 4 := by
+  unfold Aligned at *
+  simp_all
+  bv_decide
+
+@[state_simp_rules]
+theorem CheckSPAlignment_BitVecAdd_64_4 (st : ArmState) (y : BitVec 64)
+  (x_aligned : CheckSPAlignment st)
+  (y_aligned : Aligned y 4)
+  :
+  Aligned ((r (StateField.GPR 31#5) st) + y) 4 := by
+  simp_all only [CheckSPAlignment, read_gpr, zeroExtend_eq]
+  generalize (r (StateField.GPR 31#5) st) = x at *
+  apply Aligned_BitVecAdd_64_4 x_aligned y_aligned
+
```

Comment:
```suggestion
@[state_simp_rules]
theorem CheckSPAlignment_BitVecAdd_64_4 (st : ArmState) (y : BitVec 64)
  (x_aligned : CheckSPAlignment st)
  (y_aligned : Aligned y 4)
  :
  Aligned ((r (StateField.GPR 31#5) st) + y) 4 := Aligned_BitVecAdd_64_4 x_aligned y_aligned
```

Since all of these are definitionally equal, we can directly call `Aligned_BitVecAdd_64_4` and use it to solve this goal.

Reply:
Actually, after reviewing the rest of the file, I think this theorem is un-necessary even, since it's defeq.

Reply:
It's likely unnecessary.  I was following a similar pattern Shilpi had from some work I have uncommitted on `popcount32`.